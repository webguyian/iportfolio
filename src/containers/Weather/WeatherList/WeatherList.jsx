import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'components/Icon/Icon';
import Link from 'components/Link/Link';
import Text from 'components/Text/Text';

import { DEFAULT_LOCATIONS } from 'modules/weather/constants';
import { getTemp } from 'modules/weather/helpers';
import { useWeatherLocations } from 'modules/weather/hooks';

const WeatherList = props => {
  const { current } = props;
  const locations = useWeatherLocations(DEFAULT_LOCATIONS);
  const baseClass = 'weather-list';

  return (
    <ul className={baseClass}>
      {current && (
        <li className={`${baseClass}-item`}>
          <Link to="/weather">
            <Text>
              {current.city} <Icon name="location-arrow" />
            </Text>
            <Text>{getTemp(current)}</Text>
          </Link>
        </li>
      )}
      {locations.map(location => (
        <li key={location.city} className={`${baseClass}-item`}>
          <Link
            to={{
              pathname: '/weather',
              state: {
                data: location
              }
            }}
          >
            <Text>{location.city}</Text>
            <Text>{getTemp(location)}</Text>
          </Link>
        </li>
      ))}
    </ul>
  );
};

WeatherList.propTypes = {
  current: PropTypes.object
};

export default WeatherList;
