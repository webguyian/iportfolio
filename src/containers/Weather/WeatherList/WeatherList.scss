.weather-list {
  @include unstyled-list;

  &-item {
    > .ui-link {
      display: flex;
      flex-flow: row nowrap;
      align-items: center;
      justify-content: space-between;
      width: 100%;
      padding: $gutter-m;
      border-bottom: $border;
      color: $white;

      &:hover,
      &:focus {
        color: $yellow;
      }

      > .ui-text {
        font-size: $font-size-l;
      }
    }

    .ui-icon-location-arrow {
      margin-top: 4px;
      font-size: $font-size-xs;
      vertical-align: top;
    }
  }
}
