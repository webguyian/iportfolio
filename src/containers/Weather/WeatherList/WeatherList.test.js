import React from 'react';
import { create } from 'react-test-renderer';

import WeatherList from './WeatherList';

describe('<WeatherList />', () => {
  // Override prop error
  console.error = jest.fn();

  it('renders correctly', () => {
    const component = create(<WeatherList />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with current city', () => {
    const current = {
      city: 'New York City',
      currently: {
        temperature: 43
      }
    };
    const component = create(<WeatherList current={current} />);

    expect(component).toMatchSnapshot();
  });
});
