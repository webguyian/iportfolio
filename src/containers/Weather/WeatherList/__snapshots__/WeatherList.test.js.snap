// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherList /> renders correctly 1`] = `
<ul
  className="weather-list"
>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Austin",
              "coordinates": Object {
                "lat": 30.2672,
                "lon": -97.7431,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Austin
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Cupertino",
              "coordinates": Object {
                "lat": 37.323,
                "lon": -122.0322,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Cupertino
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Orlando",
              "coordinates": Object {
                "lat": 28.5383,
                "lon": -81.3792,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Orlando
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Philadelphia",
              "coordinates": Object {
                "lat": 39.9526,
                "lon": -75.1652,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Philadelphia
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Juneau",
              "coordinates": Object {
                "lat": 58.3019,
                "lon": -134.4197,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Juneau
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
</ul>
`;

exports[`<WeatherList /> renders correctly with current city 1`] = `
<ul
  className="weather-list"
>
  <li
    className="weather-list-item"
  >
    <Link
      to="/weather"
    >
      <span
        className="ui-text ui-text--body"
      >
        New York City
         
        <i
          className="ui-icon ui-icon-location-arrow fas fa-location-arrow"
        />
      </span>
      <span
        className="ui-text ui-text--body"
      >
        43Â°
      </span>
    </Link>
  </li>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Austin",
              "coordinates": Object {
                "lat": 30.2672,
                "lon": -97.7431,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Austin
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Cupertino",
              "coordinates": Object {
                "lat": 37.323,
                "lon": -122.0322,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Cupertino
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Orlando",
              "coordinates": Object {
                "lat": 28.5383,
                "lon": -81.3792,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Orlando
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Philadelphia",
              "coordinates": Object {
                "lat": 39.9526,
                "lon": -75.1652,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Philadelphia
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
  <li
    className="weather-list-item"
  >
    <Link
      to={
        Object {
          "pathname": "/weather",
          "state": Object {
            "data": Object {
              "city": "Juneau",
              "coordinates": Object {
                "lat": 58.3019,
                "lon": -134.4197,
              },
            },
          },
        }
      }
    >
      <span
        className="ui-text ui-text--body"
      >
        Juneau
      </span>
      <span
        className="ui-text ui-text--body"
      />
    </Link>
  </li>
</ul>
`;
